test/regress/regress0/arrays/issue3814.smt2

```diff
@@ -0,0 +1,12 @@
+; REQUIRES: proof
+; COMMAND-LINE: --produce-unsat-cores
+; EXPECT: sat
```

Comment:
sat and unsat core?

Reply:
Yeah, it is a bit funny. The input that the issue was reported on is actually `sat`. The assertion failure ocured during bookkeeping that we do when we have the production of unsat cores enabled (they used `--check-unsat-cores`). Rather than trying to come up with an example that triggers the bug, I've opted to reuse the original example and simply enabling unsat core production.